packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -127,32 +127,16 @@ contract MerkleDistributor is Ownable {
      *
      ****************************/
 
-    // Batch claims for a reward currency for an account to save gas. We only allow
-    // batching the same reward token and the same account because this allows us to effect
-    // the most gas optimizations for the user by precomputing the total amount of the
-    // chosen reward currency to send to the user in a single transfer transaction.
-    // If we allowed multiple accounts or multiple reward currencies,
-    // then this function would still reduce to multiple ERC20.transfer calls.
-    function claimMulti(
-        Claim[] memory claims,
-        address rewardToken,
-        address account
-    ) external {
-        uint256 amountToClaim = 0;
+    // Batch claims to reduce gas versus individual submitting all claims.
+    function claimMulti(Claim[] memory claims) external {
         for (uint256 i = 0; i < claims.length; i++) {
-            Claim memory claim = claims[i];
-            require(claim.account == account, "Invalid account in batch claim");
-            address _rewardToken = address(merkleWindows[claim.windowIndex].rewardToken);
-            require(_rewardToken == rewardToken, "Invalid rewardToken in batch claim");
-            _verifyAndMarkClaimed(claim);
-            amountToClaim = amountToClaim.add(claim.amount);
+            claim(claims[i]);
```

Comment:
personally, I prefer this implementation over the one presented in [2677](https://github.com/UMAprotocol/protocol/pull/2677/files). it's easier to reason about and cleaner. downside is more token transfers. However, from your tests it looks like it uses _less_ gas, which is quite surprising!

Reply:
Also [this test from 2677](https://github.com/UMAprotocol/protocol/pull/2677/files#r591517458) is the most realistic IMO (1 account, many trees, many reward currencies) shows that this implementation performs better than #2677

Reply:
how is this possible?

Reply:
Because #2677 only gives you gas-reductions if some of the claims re-use the same {account, rewardToken} combo, which this test did not, so it was the worst case test for #2677